relay-server/src/utils/managed_envelope.rs

```diff
@@ -305,6 +306,18 @@ impl ManagedEnvelope {
         self.envelope().meta()
     }
 
+    /// Returns estimated size of this envelope.
+    ///
+    /// This is just an estimated size, which in reality can be somewhat bigger, depending on the
+    /// list of additional attributes allocated on all of the inner types.
+    pub fn estimated_size(&self) -> usize {
+        // Always round it up to next 100 bytes.
+        (f64::ceil(
+            (self.context.summary.payload_size + size_of::<Self>() + size_of::<Envelope>()) as f64
```

Comment:
Isn't `EnvelopeContext` what you want to add here? The size of the envelope is already computed as part of the `context.summary.payload_size`, so IIUC we miss the rest of the type?
```suggestion
            (self.context.summary.payload_size + size_of::<Self>() + size_of::<EnvelopeContext>()) as f64
```

Reply:
Nope. 
The `EnvelopeContext` should already in included in `Self` at leat all the stack allocated portion, and since the `Envelope` is in the `Box`, we count only the pointer size but not the stack fields. That's why additionally to this I add the envelope here as well. 
Same goes about payload size from summary , which calculated for every time in the envelope. That's also in practice often escapes to heap (I assume 😄 from what I can see). That why added here payload size as external field summary.

Reply:
Ohh I see, that makes sense!